File: median.py
# Define a function to find the median og given integers in the sorted array
def find_median(nums):
    n = len(nums)
    if n % 2 == 0:
        # If the length of the array is even, return the average of the two middle elements
        return (nums[n // 2 - 1] + nums[n // 2]) / 2
    else:
        # If the length of the array is odd, return the middle element
        return nums[n // 2]

def main():
    nums = input("Enter the sorted array of integers separated by space: ").split()
    nums = [int(num) for num in nums]
    median = find_median(nums)
    print("The median of the array is:", median)

if __name__ == "__main__":
    main()

File: median.rs
use std::io;

// Define a function to find the median og given integers in the sorted array
fn find_median(arr: &[i32]) -> f64 {
    let n = arr.len();
    if n % 2 == 0 {
        let mid_right = arr[n / 2];
        let mid_left = arr[n / 2 - 1];
        return (mid_left as f64 + mid_right as f64) / 2.0;
    } else {
        return arr[n / 2] as f64;
    }
}

fn main() {
    // Input sorted array from the user
    println!("Enter the sorted array of integers separated by spaces:");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read input");
    let arr: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|x| x.parse().expect("Invalid input"))
        .collect();

    // Find the median of the array
    let median = find_median(&arr);

    // Print the result
    println!("Median of the array: {}", median);
}
